### PR TITLE
HW2_Spreadsheet code review branch

### PR DESCRIPTION
**Pull request by liangnaiyun-tw**
HW2_Spreadsheet code review branch

### CODE REVIEW

File: HW2_Spreadsheet/Team1/SpreadsheetApplication.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        }
+        this.presentations.add(p);
+    }
+
+    public void addData(String newItem, int newValue){
+        if (this.data == null) {
+            this.data = new LinkedHashMap<>();    
+        }
+        this.data.put(newItem, newValue);
+        this.totalItemValue += newValue;
```
Comment (container513) [inline HW2_Spreadsheet/Team1/SpreadsheetApplication.java:51]: "change"指令,後面的item可能是已存在的.
所以totalItemValue要先減原先的數量, 在加newValue.

Comment (shaoyuchu) [reply HW2_Spreadsheet/Team1/SpreadsheetApplication.java:51]: Another approach is to compute the total value in PieChart.depict(). In this case, we do not have to maintain the total value here. Both approaches are fine, it's up to you to decide :)

**Review by shaoyuchu — COMMENTED** [review]
nitpick (non-blocking): some indentations are missing in the switch-case blocks in `Main.java` and `SpreadsheetApplication.java`

**Review by container513 — APPROVED** [review]

**Review by shaoyuchu — APPROVED** [review]

File: HW2_Spreadsheet/Team1/SpreadsheetApplication.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+
+public class SpreadsheetApplication {
+
+    private LinkedHashMap<String, Integer> data;
+    private List<Presentation> presentations;
+    public static int totalItemValue = 0;
```
[the remark this answers is not in the document]
Comment (liangnaiyun-tw) [reply HW2_Spreadsheet/Team1/SpreadsheetApplication.java:9]: OK, I will edit it.

File: HW2_Spreadsheet/Team1/SpreadsheetApplication.java [hunk 3]
```diff
@@ -49,13 +49,10 @@ public void addData(String newItem, int newValue){
         }
         
         if(data.containsKey(newItem)){
-            int oldValue = data.get(newItem);
             data.remove(newItem);
```
Comment (shaoyuchu) [inline HW2_Spreadsheet/Team1/SpreadsheetApplication.java:52]: I'm not sure if the change command command modifies the items' order.
Take the following input for example:
```
// ignore the addChart commands
data Apple 8
data Orange 2
change Spreadsheet Apple 5
```
**Case 1: the change command does NOT modify the item' order**
The resulting order should be (1) Apple 5 (2) Orange 2. In this case, we should directly overwrite the entry without the remove command.
**Case 2: the change command does modify the item' order**
The resulting order should be (1) Orange 2 (2) Apple 5. In this case, we do have to remove before putting a new entry.

Comment (shaoyuchu) [reply HW2_Spreadsheet/Team1/SpreadsheetApplication.java:52]: I personally think it should be case 1. Does anyone have any idea?

Comment (liangnaiyun-tw) [reply HW2_Spreadsheet/Team1/SpreadsheetApplication.java:52]: I also think it's Case 1.
Only the value will be changed, not removed and rejoined.
I will fix it.

File: HW2_Spreadsheet/Team1/Main.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+                        break;
+                    case "addChart":
+                        spreadsheetApplication.addPresentation(keyValuePair[1]);
+                        break;
+                    case "change":
+                        System.out.println(String.format("%s change %s %s.", keyValuePair[1], keyValuePair[2], keyValuePair[3]));
+                        spreadsheetApplication.update(keyValuePair[2], Integer.parseInt(keyValuePair[3]));
+                        break;
+                    default:
+                        break;
```
Comment (container513) [inline HW2_Spreadsheet/Team1/Main.java:25]: In default, I think need to throw an exception to print "Input Error".

Comment (container513) [reply HW2_Spreadsheet/Team1/Main.java:25]: But if you throw an exception, while loop will break, so maybe you can reference Main.java of hw1.

Comment (liangnaiyun-tw) [reply HW2_Spreadsheet/Team1/Main.java:26]: OK, I will fix it

**Comment by Jackychung0522** [comment]
great

**Review by Jackychung0522 — APPROVED** [review]

**Pull request merged**